api/local_analytics_exporter.py

```diff
@@ -0,0 +1,108 @@
+import csv, codecs, cStringIO
+from StringIO import StringIO
+
+from sqlalchemy.sql import func
+from sqlalchemy.orm import lazyload
+
+from core.model import (
+    CirculationEvent,
+    Edition,
+    Genre,
+    Identifier,
+    LicensePool,
+    Work,
+    WorkGenre,
+)
+
+class LocalAnalyticsExporter(object):
+    def export(self, _db, start, end):
+
+        query = _db.query(
+                CirculationEvent, Identifier, Work, Edition
+            ) \
+            .join(LicensePool, LicensePool.id == CirculationEvent.license_pool_id) \
+            .join(Identifier, Identifier.id == LicensePool.identifier_id) \
+            .join(Work, Work.id == LicensePool.work_id) \
+            .join(Edition, Edition.id == Work.presentation_edition_id) \
+            .filter(CirculationEvent.start >= start) \
+            .filter(CirculationEvent.start < end) \
+            .order_by(CirculationEvent.start.asc())
+        query = query \
+            .options(lazyload(Identifier.licensed_through)) \
+            .options(lazyload(Work.license_pools))
+        results = query.all()
+
+        work_ids = map(lambda result: result[2].id, results)
+
+        subquery = _db \
+            .query(WorkGenre.work_id, Genre.name) \
+            .join(Genre) \
+            .filter(WorkGenre.work_id.in_(work_ids)) \
+            .order_by(WorkGenre.affinity.desc()) \
+            .subquery()
+        genre_query = _db \
+            .query(subquery.c.work_id, func.string_agg(subquery.c.name, ",")) \
+            .select_from(subquery) \
+            .group_by(subquery.c.work_id)
+        genres = dict(genre_query.all())
+
+        header = [
+            "time", "event", "identifier", "identifier_type", "title", "author",
+            "fiction", "audience", "publisher", "language", "target_age", "genres"
+        ]
+
+        def result_to_row(result):
+            (event, identifier, work, edition) = result
+            return [
+                str(event.start) or "",
+                event.type,
+                identifier.identifier,
+                identifier.type,
+                edition.title,
+                edition.author,
+                "fiction" if work.fiction else "nonfiction",
+                work.audience,
+                edition.publisher,
+                edition.language,
+                work.target_age_string,
+                genres.get(work.id)
+            ]
+
+        data = [header] + map(result_to_row, results)
+
+        output = StringIO()
+        writer = UnicodeWriter(output)
+        writer.writerows(data)
+        return output.getvalue()
+
+
+class UnicodeWriter:
```

Comment:
Not necessary to make this change now, but I discovered there's a unicodecsv package that works really well.